src/arrayMethodSort.js

```diff
@@ -4,8 +4,29 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    let conditionFunction = compareFunction;
+
+    if (!conditionFunction) {
+      conditionFunction = function (a, b) {
+        return String(a) > String(b) ? 1 : -1;
+      };
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      for (let j = i + 1; j < this.length; j++) {
+        const a = this[i];
+        const b = this[j];
+        const result = conditionFunction(a, b);
+
+        if (result > 0) {
+          this[j] = a;
+          this[i] = b;
+        }
+      }
+    }
```

Comment:
This is a performance issue. You are using a bubble sort algorithm, which has an average and worst-case performance of O(n^2). For large arrays, this could lead to performance issues. You might want to consider using a more efficient sorting algorithm like quicksort or mergesort. However, if you are sure that this function will only be used for small arrays, bubble sort is fine.